### PR TITLE
try to handle ConnectionReset error in windows CI for unit tests

### PR DESCRIPTION
**Pull request by bartlomieju**
Ref #4373

### CODE REVIEW

File: cli/js/tests/unit_test_runner.ts [hunk 1]
```diff
@@ -127,31 +127,50 @@ async function runTestsForPermissionSet(
...
+  } catch (e) {
+    err = e;
+  } finally {
+    // Close socket to worker, it should shutdown gracefully.
+    conn.close();
   }
 
-  // Close socket to worker, it should shutdown gracefully.
-  conn.close();
+  if (typeof err !== "undefined") {
```
Comment (ry) [inline cli/js/tests/unit_test_runner.ts]: Is this necessary? Can't it just be `if (err instanceof Deno.errors.ConnectionReset) {` ?

File: cli/js/tests/unit_test_runner.ts [hunk 2]
```diff
@@ -127,31 +127,50 @@ async function runTestsForPermissionSet(
...
+    // There are intermittent `ConnectionReset` errors occurring
+    // on Windows CI; if error occurred but test runner
+    // sent end message then the error is ignored.
```
Comment (ry) [inline cli/js/tests/unit_test_runner.ts]: I think ConnectionReset is always possible in sockets, and just needs to be handled. The comment doesn't need to talk about the flakiness.

File: cli/js/tests/unit_test_runner.ts [hunk 3]
```diff
@@ -127,31 +127,50 @@ async function runTestsForPermissionSet(
...
+    if (err instanceof Deno.errors.ConnectionReset) {
+      // If we didn't receive end event it's an actual error
+      if (typeof endEvent === "undefined") {
```
Comment (ry) [inline cli/js/tests/unit_test_runner.ts]: `if (!endEvent)` is sexier

**Comment by bartlomieju** [comment]
> Assuming those last two go green, LGTM

Error still occurs... I added `--verbose` flag to see what's the output from worker when error occurs

File: cli/js/tests/unit_test_runner.ts [hunk 4]
```diff
@@ -127,31 +127,46 @@ async function runTestsForPermissionSet(
...
+  } catch (e) {
+    err = e;
+  } finally {
+    // Close socket to worker, it should shutdown gracefully.
+    conn.close();
   }
 
-  // Close socket to worker, it should shutdown gracefully.
-  conn.close();
+  if (err) {
```
Comment (ry) [inline cli/js/tests/unit_test_runner.ts]: this conditional seems superfluous

Comment (bartlomieju) [reply cli/js/tests/unit_test_runner.ts]: Why? Error is not thrown for every run; it can still be undefined

**Pull request merged**